cli/config.py

```diff
@@ -398,16 +399,37 @@ def merge(self, id, job_config_yaml=None):
                     "No path to application specified for {} in job.yaml".format(entry['name']))
 
 
+def get_file_if_exists(file, local: bool):
+    if local:
+        if os.path.exists(os.path.join(aztk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, file)):
+            return os.path.join(aztk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, file)
+    else:
+        if os.path.exists(os.path.join(aztk.utils.constants.GLOBAL_CONFIG_PATH, file)):
+            return os.path.join(aztk.utils.constants.GLOBAL_CONFIG_PATH, file)
+
+
 def load_aztk_spark_config():
-    def get_file_if_exists(file, local: bool):
-        if local:
-            if os.path.exists(os.path.join(aztk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, file)):
-                return os.path.join(aztk.utils.constants.DEFAULT_SPARK_CONF_SOURCE, file)
-        else:
-            if os.path.exists(os.path.join(aztk.utils.constants.GLOBAL_CONFIG_PATH, file)):
-                return os.path.join(aztk.utils.constants.GLOBAL_CONFIG_PATH, file)
+    spark_defaults_conf = spark_env_sh = core_site_xml = None
+
+    spark_defaults_conf = get_file_if_exists('spark-defaults.conf', False)
+    spark_env_sh = get_file_if_exists('spark-env.sh', False)
+    core_site_xml = get_file_if_exists('core-site.xml', False)
+
```

Comment:
I'm not sure I understand why you're making the same 3 calls with False, then True - seems kind of strange to me. Is this mean to override the defaults?

Reply:
Agreed this is poorly written, but what it's doing is getting the global configuration file, and then overwriting it with the local configuration file if it exists. I will rewrite this before I merge -- too clunky as is.